khelo/lib/components/confirmation_dialog.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:khelo/domain/extensions/context_extensions.dart';
+import 'package:style/extensions/context_extensions.dart';
+
+void showConfirmationDialog(
+  BuildContext context, {
+  String? title,
+  String? message,
+  String? confirmBtnText,
+  bool isDestructiveAction = true,
+  String? cancelBtnText,
+  required VoidCallback onConfirm,
+  VoidCallback? onCancel,
+}) {
+  showAdaptiveDialog(
+    context: context,
+    barrierDismissible: true,
+    builder: (context) {
+      return AlertDialog.adaptive(
+        surfaceTintColor: context.colorScheme.containerNormalOnSurface,
+        title: (title != null) ? Text(title) : null,
+        content: Text(message ?? context.l10n.alert_confirm_default_title),
+        actions: [
+          adaptiveAction(
+            context: context,
+            onPressed: () async {
+              context.pop();
+              onCancel?.call();
+            },
+            child: Text(cancelBtnText ?? context.l10n.common_cancel_title),
+          ),
+          adaptiveAction(
+            context: context,
+            isDestructiveAction: isDestructiveAction,
+            onPressed: () async {
+              context.pop();
+              onConfirm();
+            },
+            child: Text(
+              confirmBtnText ?? context.l10n.common_yes_title,
+            ),
+          ),
+        ],
+      );
+    },
+  );
+}
+
+Widget adaptiveAction({
+  required BuildContext context,
+  required VoidCallback onPressed,
+  required Widget child,
+  bool isDestructiveAction = false,
+}) {
+  final ThemeData theme = Theme.of(context);
+  switch (theme.platform) {
+    case TargetPlatform.android:
+    case TargetPlatform.fuchsia:
+    case TargetPlatform.linux:
+    case TargetPlatform.windows:
+      return TextButton(
+        onPressed: onPressed,
+        child: DefaultTextStyle(
+          style: TextStyle(
+            inherit: true,
+            color: isDestructiveAction
+                ? context.colorScheme.alert
+                : context.colorScheme.textSecondary,
+          ),
+          child: child,
+        ),
+      );
+    case TargetPlatform.iOS:
+    case TargetPlatform.macOS:
+      return CupertinoDialogAction(
+        onPressed: onPressed,
+        isDestructiveAction: isDestructiveAction,
+        textStyle: isDestructiveAction
+            ? null
+            : TextStyle(
+                color: isDestructiveAction
+                    ? context.colorScheme.textPrimary
+                    : context.colorScheme.textSecondary,
+              ),
+        child: child,
+      );
+  }
```

Comment:
Do we really need this? aren't Action buttons adaptive to Platform by default?
I am just asking for information. it's nice to design UI according to the  Platform

Reply:
nope, if we use textButton in ios with adaptive dialog, it will show splash when we tap on it and that looks bad, even in their official doc they have implemented it like this.

[AlertDialog.adaptive](https://api.flutter.dev/flutter/material/AlertDialog/AlertDialog.adaptive.html)